### PR TITLE
Initial GH actions unit tests for crypto modules

### LINKED ISSUE #14241: Initial GH actions unit test for crypto modules

### PR DESCRIPTION
**Pull request by mposolda**
Closes #14241

- Adding crypto modules test with BCFIPS provider to GH actions
-  The tests in the module `crypto/fips1402` are executed 2 times: once in BC approved mode and once in non-approved mode (approved mode is more strict in what algorithms and ciphers are allowed)
- For now, GH actions are triggered on the regular ubuntu host (non special FIPS-configured server). However it uses java security file, which overrides the security providers configuration to test just with the "Secure" java providers. For now, it tests just withe the BCFIPS providers as even on FIPS enabled RHEL 8 OpenJDK 11, there are some unsecured providers allowed by default (EG. SUN provider) and hence it is by default not guaranteed that you use only "secure" cryptography.

### CODE REVIEW

**Comment by mposolda** [comment]
@stianst Do you please have an opportunity to review this?

**Review by pedroigor — APPROVED** [review]
Nice :)

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -90,6 +90,50 @@ jobs:
...
+        run: mkdir -p ~/.m2 ; cp .github/settings.xml ~/.m2/
+      - name: Cleanup org.keycloak artifacts
+        run: rm -rf ~/.m2/repository/org/keycloak >/dev/null || true
+      - name: Download built keycloak
+        id: download-keycloak
+        uses: actions/download-artifact@v3
+        with:
+          path: ~/.m2/repository/org/keycloak/
+          name: keycloak-artifacts.zip
+      - name: Run crypto tests (BCFIPS non-approved mode)
```
Comment (mabartos) [inline .github/workflows/ci.yml:114]: @mposolda Great! Just nitpick regards optimization.
Is it possible to use the matrix for the approved/non-approved mode in order to execute those tests in parallel? For instance, the base testsuite uses that.

Comment (mposolda) [reply .github/workflows/ci.yml:114]: @mabartos Thanks for the feedback. I am aware of the matrix, but in this case it looked to me like unnecessary overhead as the unit tests itself are very fast to run (usually 10-15 seconds) and hence the matrix will be rather overhead due the pre-steps (like "Download built Keycloak"), which themselves need to be run for each part of the matrix independently.

Maybe we can consider add matrix later if the unit tests grow to the level when they take more significant time to run, however not sure if it happens for the "crypto" modules. WDYT?

Comment (mabartos) [reply .github/workflows/ci.yml:114]: @mposolda You're right, agree. Thanks!

**Review by mabartos — APPROVED** [review]

**Comment by mposolda** [comment]
@pedroigor @mabartos Thanks for the review!

**Pull request merged**